tests/e2e/utils/data_fetcher/logs.py

```diff
@@ -143,6 +146,11 @@ def get_logs(
     LogFetcher
         LogFetcher instance with logs available as properties and methods
     """
-    return LogFetcher(
+    log_fetcher = LogFetcher(
         function_name=function_name, start_time=start_time, filter_expression=filter_expression, log_client=log_client
     )
+
+    if expected_number_of_logs is not None and len(log_fetcher) < expected_number_of_logs:
```

Comment:
after that change, this is simplified with and easier to read with:

```suggestion
    if minimum_log_entries < len(log_fetcher):
```